### PR TITLE
replace auto_ptr with unique_ptr

### PR DESCRIPTION
**Pull request by w-vi**
Updates:
- Catch
- markdon parser
- travis CI
- C++11 flags in common.gypi

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,11 +12,5 @@ before_install:
   - rvm use $RVM --install --binary --fuzzy
   - gem update --system
   - gem --version
```
Comment (pksunkara) [inline .travis.yml]: Please remove this ruby related stuff.

File: .travis.yml [hunk 2]
```diff
@@ -10,11 +12,5 @@ before_install:
...
-notifications:
-  email:
-    recipients:
-      - z@apiary.io
-    on_success: change
-    on_failure: always
 env:
   - RVM=2.0.0
```
Comment (pksunkara) [inline .travis.yml]: Ruby stuff.

**Comment by pksunkara** [comment]
You also need to update provisioning.sh. I don't know if we need to update vcbuild.bat with C++11 flag though.

**Comment by w-vi** [comment]
what is the provision.sh good for?

As you can see appveyor works without any changes, there is nothing like  `-std=c++11` in msvc so there is nothing to do.

**Comment by w-vi** [comment]
The Ruby stuff can be removed in different PR, it is not related.

**Comment by pksunkara** [comment]
> provision.sh is used with vagrant to provision the VM box so it has all the tools needed to build snowcrash

Until we move to docker, this still needs to exist. Though we might need to update the compiler versions in that file and remove ruby.

**Comment by w-vi** [comment]
This will take more time than the whole update across all the repos :) I don't have vagrant installed, but I guess we can just switch to Ubuntu trusty and be done with it no need for this script anymore.

**Comment by pksunkara** [comment]
I am not sure if it works that way. You have to talk to @zdne before removing it.

**Pull request merged**